astropy/timeseries/binned.py

```diff
@@ -119,6 +121,11 @@ def __init__(self, data=None, *, time_bin_start=None, time_bin_end=None,
         if isinstance(time_bin_size, TimeDelta):
             time_bin_size = time_bin_size.sec * u.s
 
+        # If any of the parameter 'time_bin_start' or 'time_bin_end' is not scalar, a warning is raised.
+        if n_bins:
```

Comment:
Here, some of the tests are raising the error `AttributeError: 'NoneType' object has no attribute 'isscalar'`. This is because while ``time_bin_start`` is always required (which means it will not be `None`), ``time_bin_end`` is not necessary to be provided - ``time_bin_size`` could be provided instead. Therefore, add an extra check here as follows:

```suggestion
        if nbins is not None and time_bin_end is not None: 
            if not (time_bin_start.isscalar and  time_bin_end.isscalar):
                raise TypeError("'n_bins' is  not valid as 'time_bin_start' or 'time_bin_size'
                                              are not scalar.")
```
Note that I've changed the warning to an exception as mentioned in comment [11262#issuecomment-765434604](https://github.com/astropy/astropy/issues/11262#issuecomment-765434604).